scripts/justmat/foulplay.lua

```diff
@@ -59,62 +90,106 @@ local reset = false
 local view = 0     -- 0 == home, 1 == track edit, 2 == global edit 
 local page = 0
 local track_edit = 1
-local stopped = 1 -- change this to 0 if you would like to start running
+local stopped = 1 
+-- added for grid support - junklight 
+local currentmemcell = 1
+local currentmemcell_x = 4
+local currentmemcell_y = 1
+local copymode = false
+local blink = false
+local copysource_x = -1
+local copysource_y = -1
+-- mutes are global 
+-- grid column 2 and tracks 1-8
+local mutes = {} 
+for i=1,8 do
+  mutes[i] = false
+end
+
+
+function simplecopy(obj)
```

Comment:
aside: we should have a util for shallow copies somewhere.

maybe we do already?

/cc @catfact @tehn

Reply:
another aside: i notice that some functions are local and others are not.

curious if there's a rationale for that?

/cc @junklight